src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -376,7 +393,15 @@ private BoundExpression VisitCollectionBuilderCollectionExpression(BoundCollecti
             Debug.Assert(spanType.OriginalDefinition.Equals(_compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T), TypeCompareKind.AllIgnoreOptions));
 
             var elementType = spanType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics[0];
-            BoundExpression span = VisitArrayOrSpanCollectionExpression(node, CollectionExpressionTypeKind.ReadOnlySpan, spanType, elementType);
+
+            // If collection expression is of form `[.. anotherReadOnlySpan]`
+            // with `anotherReadOnlySpan` being a ReadOnlySpan of the same type as target collection type
+            // and that span cannot be captured in a returned ref struct
+            // we can directly use `anotherReadOnlySpan` as collection builder argument and skip the copying assignment.
+            var canSkipCopyingArgument = !constructMethod.ReturnType.IsRefLikeType || constructMethod.Parameters[0].EffectiveScope == ScopedKind.ScopedValue;
+            BoundExpression span = canSkipCopyingArgument && IsSingleReadOnlySpanSpread(node, elementType, out var spreadExpression)
```

Comment:
>IsSingleReadOnlySpanSpread(node, elementType, out var spreadExpression) [](http://example.com/codeflow?start=61&length=71)

Consider checking `IsSingleReadOnlySpanSpread()` before checking `constructMethod.ReturnType` or `constructMethod.Parameters` since in most cases, `IsSingleReadOnlySpanSpread` will return `false`.